collector/pkg/model/attribute_map.go

```diff
@@ -17,6 +17,14 @@ type AttributeMap struct {
 	values map[string]AttributeValue
 }
 
+func (a AttributeMap) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.values)
+}
+
+func (a AttributeMap) UnmarshalJSON(data []byte) error {
+	return json.Unmarshal(data, &a.values)
+}
```

Comment:
Why not use the pointer as the function receiver?

Reply:
In example of [encoding/json](https://pkg.go.dev/encoding/json) ,  `MarshalJSON()` should use the obj as the receiver ,and `UnmarshalJSON(b []byte)` should use the pointer as the receiver . 

We usually call `json.Unmarshal([]bytes, &obj)` and `json.Marshal(obj)` to invoke functions above ,I guess that's the reason. I will change the receiver of `UnmarshalJSON()`'

``` go
func (a *Animal) UnmarshalJSON(b []byte) error {
	...
}

func (a Animal) MarshalJSON() ([]byte, error) {
	...
}
```